### ISSUE TITLE
Add new Scrollable Search Panel display option to SSS

### ISSUE THREAD

**Issue by dhilditch** [superspeedyplugins/issue-tracker#146]
1. Add a new option to the SSS Ajax tab to let users switch to the scrollable search panel. This should be on by default for new users and off by default for existing users (if they already have options)
2. Create an improved HTML container system for the ajax panel that gets used when the option above is on.

Some notes:

1. Suggestions should appear first using max width of 300px if enabled
2. Taxonomy results should appear to the right with 300px width if there is space, otherwise underneath the suggestions
3. Then each post type should be listed in their CORRECT CONFIGURED ORDER except:
3a) Fetch 20 items, or however many is normal for a full search results page
3b) Let the user scroll left or right with arrows and/or touch on tablets and phones
3c) This makes the search box a consistent height - I want to eliminate the vertical scroll bar, so heights of sections could be based on CSS vh for viewheight?

The aim here is to get the basics of what we discussed in that google doc made so we can play with it.

The extra aim is to avoid layout shift that happens when post type queries arrive late. With the approach above, users can say posts then KBs then products and they are always guaranteed to be returned in that order.

You must make sure all the existing keyboard control system that is in place continues to work with this new panel and you also must ensure that the old search panel display still works.

**Comment by twarrdy** [comment]
It would be nice if we could override template and easily adjust all aspects of your component with direct access to the source code. There are too many ways to customize the user interface to restrict its capabilities with one highly opinionated template.

**Comment by dhilditch** [comment]
@twarrdy agreed. To clarify, the new template should exist in the templates folder AND the template finder should check the users theme folder first so they can override the template. That's also currently possible right now for grid/list display options, so the new template should follow that so users can edit it if they need to.

However, in this case, it would also be good to have the actual container be in a template file too - currently the container for search is hard coded in the PHP and then that container calls the grid or list template for each item. This new setup should use a template file for the container (search box and dropdown panel to contain everything) and then it can use the existing grid and list templates to display each item in the search results.

Add some comments throughout the container template file so that advanced users can figure out quickly what they can change and what they must keep.

![image](https://github.com/superspeedyplugins/issue-tracker/assets/3008613/a4430bb3-1715-4a08-b2d8-e3d1c0af6135)

If you need help with this, we can work on it together. The first thing I'd like to see is the template container and really the existing options can be retained - the existing options give the enabled post types and the order of post types, and the display template (list or grid) so we would just be ignoring the 'Num Items'.

**Comment by twarrdy** [comment]
It would be helpful to have two additional ajax templates:

1) no results template
2) on focus template - on/off - on focus, the ajax search panel appears and displays search history/trending searches/custom code etc.

They can be empty for now.

**Comment by dhilditch** [comment]
![image](https://github.com/superspeedyplugins/issue-tracker/assets/3008613/e4c35601-2c27-448a-95cf-87d989c5bc21)

// appearing in the request, please fix this and make it single slash

This above means that the check to see if we can use our mu-plugin fails.

These new endpoints you have created, you will have to mirror them in the mu plugin override I created for our Ultra Ajax. 

![image](https://github.com/superspeedyplugins/issue-tracker/assets/3008613/2a08fa26-f61b-4031-a29b-8ceb9781bab9)
In screenshot above you can see the 2 command overrides in our ultra ajax (mu plugin) override.

(suggestions and posts can use ultra ajax currently)

But in your code, you added a new sub-command, so this will break our ultra ajax.

Please make it compatible.

![image](https://github.com/superspeedyplugins/issue-tracker/assets/3008613/a95b9a13-5593-4939-9719-d306e0bb2cc2)

I switched to the new panel, I added a breakpoint in the sss_scrollable_rest_getPosts function, it does not get hit.

For us, it's still using the old code.

![image](https://github.com/superspeedyplugins/issue-tracker/assets/3008613/6c769ee2-0444-45e0-98cb-9f8d49aa0d9e)

**Comment by dhilditch** [comment]
fixed in version 4.77
